### PR TITLE
Use BuildList for resourcePackageNames to remove hard coded separator

### LINKED ISSUE #346: Using non transitive resources without additional dependencies leads to  ClassNotFoundException
When the module using Paparazzi has no additional dependencies, setting `android.nonTransitiveRClass=true` in `gradle.properties` can lead to this error: `java.lang.ClassNotFoundException: /R`

Turns out that this code 

https://github.com/cashapp/paparazzi/blob/4890fdc31d266f68e8e79e21c6a879064d590a99/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt#L74-L80

adds a "," after the `mainPackage` even if `artifactFiles` is empty, causing an empty string to be returned when splitting the string here

https://github.com/cashapp/paparazzi/blob/4890fdc31d266f68e8e79e21c6a879064d590a99/paparazzi/src/main/java/app/cash/paparazzi/Environment.kt#L54

The returned list looks like this `["$mainPackage", ""]`

I'll add a potential fix and test cases later.

### PR DESCRIPTION
**Pull request by LuK1709**
Resolves #346

### CODE REVIEW

File: paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt [hunk 1]
```diff
@@ -66,15 +66,19 @@ abstract class PrepareResourcesTask : DefaultTask() {
   private val projectDirectory = project.layout.projectDirectory
 
   @TaskAction
+  @OptIn(ExperimentalStdlibApi::class)
```
Comment (jrodbx) [inline paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt:69]: @JakeWharton, correct if I'm wrong, but I think you had said in the past (maybe in a Timber PR?) that using experimental apis, like buildList etc. in libraries, isn't ideal.  Is that guidance still accurate?

Comment (LuK1709) [reply paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt:69]: There where already some instances of `ExperimentalStdlibApi`, one example being https://github.com/cashapp/paparazzi/blob/9fcffd16467697d85cb4518a188cc7fbfab93092/paparazzi/src/main/java/app/cash/paparazzi/internal/parsers/ResourceParser.kt#L124-L126
That's why I included it here. 

Apart from that, are there any changes required?

Comment (jrodbx) [reply paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt:69]: fair enough.  i forgot we used that elsewhere.  we can migrate later then, if an issue!

File: paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt [hunk 2]
```diff
@@ -66,15 +66,19 @@ abstract class PrepareResourcesTask : DefaultTask() {
...
     out.delete()
 
     val mainPackage = packageName.get()
     val resourcePackageNames = if (nonTransitiveRClassEnabled.get()) {
-      "$mainPackage," + artifactFiles.files.joinToString(",") { file ->
-        file.useLines { lines -> lines.first() }
-      }
+      buildList {
+        add(mainPackage)
+        addAll(artifactFiles.files.map { file ->
```
Comment (jrodbx) [inline paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt:78]: nit: if we move `add` inside the map, we avoid creation of a temp list?

File: paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt [hunk 3]
```diff
@@ -66,15 +66,19 @@ abstract class PrepareResourcesTask : DefaultTask() {
...
+          file.useLines { lines -> lines.first() }
+        })
```
Comment (jrodbx) [inline paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt:80]:
```suggestion
        artifactFiles.files.map { file ->
          add(file.useLines { lines -> lines.first() })
        }
```

Comment (ChristianBecker) [reply paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt]: Shouldn't the `map` become a `forEach` to get rid of the temp list? I just created #352 for it.

**Pull request merged**